src/types/generic.ts

```diff
@@ -8,43 +8,38 @@ export type EnvironmentVariables = {
   [key: string]: string | undefined;
 };
 
+export type FileInfo = {
+  name: string;
+  path: string;
+};
+
 /**
- * Necessary info to deploy a function or asset
+ * Necessary info to deploy a function
  */
-export type FileInfo = {
+export type ServerlessResourceConfig = {
   /**
-   * Relative file name/path. Will be used to create the Serverless paths. Example:
-   * `example.js` => `/example`
-   * `sms/reply.protected.js` => `/sms/reply`
-   * `index.html` => `/index.html`
+   * Access for the function
    */
-  name: string;
+  access: AccessOptions;
   /**
-   * Path on file system to read the contents from
+   * Content of the uploaded function
    */
-  path?: string;
+  content: string | Buffer;
   /**
-   * Alternantive to `path` if you want to specify the content directly.
-   * Can be a string or a `Buffer` instance representing the content.
+   * Function name
    */
-  content?: string | Buffer;
-};
-
-export type RawFunctionWithPath = FileInfo & {
-  functionPath: string;
-  access: AccessOptions;
-};
-
-export type RawAssetWithPath = FileInfo & {
-  assetPath: string;
-  access: AccessOptions;
+  name: string;
```

Comment:
> I don't support a naming capability in the CLI because the filesystem only has the concept of file names and they turn into the path anyways

I added the naming capability because by introducing `deployProject` serverless-api opened up to be used programmatically. When developing on top of it I was reading the API docs and found myself digging into source code wondering where the name is set. My expectation was that I could set all the things the API offers (including `name`) using `deployProject`.

After reading the source code and understanding how things work together I think that the file-system logic (and using file system path as the name) belongs only to `deployLocalProject`.

As an example:
```
const result = await client.deployProject({
  env: {},
  pkgJson: {},
  serviceName: 'serverless-api-test-3',
  functionsEnv: 'dev',
  functions: [
    {
      name: 'goodbye-world',
      path: '/goodbye-world-path',
      content: await readFile(path.join(__dirname, 'src', 'handler.js')),
      access: 'public',
    },
  ],
  assets: [
    {
      name: 'booth',
      path: '/foo/booth.jpg',
      access: 'public',
      content: await readFile(path.join(__dirname, 'static', 'booth.jpg')),
    },
  ],
  overrideExistingService: true,
});
```

This is the code I expected to write and frameworks on top of serverless providers have a name concept (serverless framework has it, too), and while `serverless-api` was imagined as a tool that works with the file system I think it makes sense to keep `deployProject` agnostic. :)